tests/test_transforms.py

```diff
@@ -1428,3 +1429,37 @@ def test_randomgridshuffle():
         assert len(w) == 1
         assert w[0].category is UserWarning
     warnings.resetwarnings()
+
+
+@pytest.mark.parametrize("num_shadows_limit, num_shadows_lower, num_shadows_upper, expected_warning", [
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (testing):** Add a test case for `RandomShadow` with valid `num_shadows_limit` but without deprecated parameters.

It's good to test the deprecation warnings and the error handling for invalid parameters. However, it would also be useful to have a test case that passes a valid `num_shadows_limit` without using the deprecated parameters to ensure that the functionality works as expected in the absence of deprecation.

```suggestion
@pytest.mark.parametrize("num_shadows_limit, num_shadows_lower, num_shadows_upper, expected_warning", [
    ((1, 2), None, None, None),
    ((2, 3), None, None, None),
    ((1, 2), 1, None, DeprecationWarning),
    ((1, 2), None, 2, DeprecationWarning),
    ((1, 2), 1, 2, DeprecationWarning),
    ((2, 1), None, None, ValueError),
    ((3, 4), None, None, None),  # Valid case without deprecated parameters
])
```